src/callbacks_stage/positivity_shallow_water_dg1d.jl

```diff
@@ -84,4 +86,89 @@ function limiter_shallow_water!(u, threshold::Real, variable,
 
     return nothing
 end
+
+# Note that for the `ShallowWaterMultiLayerEquations1D` only the waterheight `h` is limited in
+# each layer. Furthermore, a velocity desingularization is applied after the limiting to avoid
+# numerical problems near dry states.
+#
+# !!! warning "Experimental code"
+#     This is an experimental feature and may change in future releases.
+function limiter_shallow_water!(u, threshold::Real, variable,
+                                mesh::Trixi.AbstractMesh{1},
+                                equations::ShallowWaterMultiLayerEquations1D,
+                                dg::DGSEM, cache)
+    @unpack weights = dg.basis
+
+    Trixi.@threaded for element in eachelement(dg, cache)
+        # Limit layerwise
+        for m in eachlayer(equations)
+            # determine minimum value
+            value_min = typemax(eltype(u))
+            for i in eachnode(dg)
+                u_node = get_node_vars(u, equations, dg, i, element)
+                value_min = min(value_min, variable(u_node, equations)[m])
+            end
+
+            # detect if limiting is necessary
+            value_min < threshold - eps() || continue
+
+            # compute mean value
+            u_mean = zero(get_node_vars(u, equations, dg, 1, element))
+            for i in eachnode(dg)
+                u_node = get_node_vars(u, equations, dg, i, element)
+                u_mean += u_node * weights[i]
+            end
+
+            # note that the reference element is [-1,1]^ndims(dg), thus the weights sum to 2
+            u_mean = u_mean / 2^ndims(mesh)
+
+            # We compute the value directly with the mean values.
+            # The waterheight `h` is limited independently in each layer.
+            value_mean = variable(u_mean, equations)[m]
+            theta = (value_mean - threshold) / (value_mean - value_min)
+
+            for i in eachnode(dg)
+                u_node = get_node_vars(u, equations, dg, i, element)
+                h_node = waterheight(u_node, equations)[m]
+                h_mean = waterheight(u_mean, equations)[m]
+
+                u[m, i, element] = theta * h_node + (1 - theta) * h_mean
+            end
+        end
+    end
```

Comment:
These lines should be covered by testing, but they are not tracked by Codecov as they are not executed in the first time step. 
Is it somehow possible to manually force a single coverage test to run for more than one iteration?

Reply:
Yes. As an argument to `@test_trixi_include`, you should be able to provide
```julia
@test_trixi_include(...; coverage_override = (; maxiters = 10))
```
which overrides certain properties just for coverage testing, in this case the `maxiters` variable, which by default is set to `1` if not provided for coverage tests, see also
https://github.com/trixi-framework/Trixi.jl/blob/dd051566a9079e22dfa792b145c0b50eaafa37fd/test/test_trixi.jl#L24-L27
and, for example,
https://github.com/trixi-framework/Trixi.jl/blob/dd051566a9079e22dfa792b145c0b50eaafa37fd/test/test_mpi_p4est_2d.jl#L80-L85

Reply:
Thanks! That should provide an easy fix for the problem :+1:

Reply:
With the increased `maxiters=15` in `elixir_shallowwater_multilayer_parabolic_bowl.jl` introduced in commit 5bda15fc4bae171ac74b9c07e3ef4d2a00783fc4 the lines are now tracked by Codecov.